apps/src/templates/courseOverview/CourseOverview.js

```diff
@@ -212,7 +224,7 @@ class CourseOverview extends Component {
         )}
         {isSignedIn && (
           <Announcements
-            announcements={this.props.announcements}
+            announcements={this.getAnnouncementsWithAnalyticsData()}
```

Comment:
Is the analytics data specific to each announcement or the same for all announcements?  (Wondering why we're splicing it into each announcement vs. passing a separate prop to Announcements.)

Reply:
mmm good point, not in this case. I guess it would make it a bit more flexible to allow for information specific to each announcement to be included, but if we want to keep it simple for the case we're using it for now I can make it it's own prop.

Reply:
@jamescodeorg prefer that I don't inject the analytics data into the announcement?

Reply:
If we had announcement-specific data that we wanted to put in the analytics, where would it come from?  Would we just put it on announcement shape?

I guess this is the thing that feels weird about this -- if the analytics data is per-announcement, shouldn't we just add it to the announcement shape (instead of injecting it in ScriptOverviewHeader)?

If we don't think that we'll have per-announcement analytics data, then I think passing it as a single prop to Announcements would be easier?

Reply:
> If we had announcement-specific data that we wanted to put in the analytics, where would it come from? Would we just put it on announcement shape?
> 
> I guess this is the thing that feels weird about this -- if the analytics data is per-announcement, shouldn't we just add it to the announcement shape (instead of injecting it in ScriptOverviewHeader)?
> 
> If we don't think that we'll have per-announcement analytics data, then I think passing it as a single prop to Announcements would be easier?

I don't have a strong preference -- I'll make analyticsData it's own prop :).